src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EventRecordRepositoryTest.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.reform.blobrouter.data;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.blobrouter.data.model.Event;
+import uk.gov.hmcts.reform.blobrouter.data.model.EventRecord;
+import uk.gov.hmcts.reform.blobrouter.data.model.NewEventRecord;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.DISPATCHED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.FILE_RECEIVED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.REJECTED;
+
+@ActiveProfiles({"integration-test", "db-test"})
+@SpringBootTest
+public class EventRecordRepositoryTest {
+
+    private static final String CONTAINER = "container";
+    private static final String FILE_NAME = "file_name";
+
+    @Autowired private EventRecordRepository repo;
+    @Autowired private DbHelper dbHelper;
+
+    @BeforeEach
+    void setUp() {
+        dbHelper.deleteAll();
+    }
+
+    @ParameterizedTest
+    @MethodSource("eventWithNotes")
+    void should_save_events_separately(Event event, String notes) {
+        // given
+        var newEnvelope = notes == null
+            ? new NewEventRecord(CONTAINER, FILE_NAME, event)
+            : new NewEventRecord(CONTAINER, FILE_NAME, event, notes);
+
+        // when
+        long id = repo.insert(newEnvelope);
+
+        // and
+        List<EventRecord> eventsInDb = repo.find(CONTAINER, FILE_NAME);
+
+        // then
+        assertThat(eventsInDb)
+            .hasSize(1)
+            .first()
+            .satisfies(record -> {
+                assertThat(record.id).isEqualTo(id);
+                assertThat(record.event).isEqualTo(event);
+
+                if (notes == null) {
+                    assertThat(record.notes).isNull();
+                } else {
+                    assertThat(record.notes).isEqualTo(notes);
```

Comment:
will this not work for null?

Reply:
nope, there's a null check in the method